runtime/io.js

```diff
@@ -138,8 +138,34 @@ function caml_ml_open_descriptor_in (fd)  {
   var refill = null;
   if(fd == 0 && fs_node_supported()){
     var fs = require('fs');
+    var Buffer = require('buffer').Buffer;
     refill = function () {
-      return caml_string_of_jsstring(fs.readFileSync(0, 'utf8'))};
+      var buf = Buffer.alloc(256);
+      // Ref https://gist.github.com/espadrine/172658142820a356e1e0
+      var stdinFd;
+      var needsClose = false;
+      try {
+        stdinFd = fs.openSync('/dev/stdin', 'rs');
```

Comment:
We we need to try reading `/dev/stdin` first ? can't we process.stdin.fd / 0 directly ?

Reply:
Yes, the fd0 on linux and mac might be async—which happens any time someone tries to read from stdin using `process.stdin` (or even access the fd property, as noted in the linked issue).

So it is much safer to open a new file descriptor `/dev/stdin` in sync mode. The fallback to fd0 is used in the try/catch because there might be weird environments where /dev/stdin doesn't exist (such as the `pkg` tool that Grain uses).